routes/index.js

```diff
@@ -1,9 +1,34 @@
-const router = require('express').Router();
+const router = require("express").Router();
 
-const cache = require('../middlewares/cachePolicy');
-const url = require('../controllers/url');
+const passport = require("passport");
+const cache = require("../middlewares/cachePolicy");
+const url = require("../controllers/url");
 
-router.post('/shorten',url.shorten);
-router.get('/:code',cache,url.redirect);
+router.post("/shorten", url.shorten);
+router.get("/:code", cache, url.redirect);
 
-module.exports = router;
\ No newline at end of file
+// auth logout
+router.get("/auth/logout", (req, res) => {
+  req.logout();
+  res.redirect("/");
+});
+
+// auth with google+
+router.get(
+  "/auth/google",
+  passport.authenticate("google", {
+    scope: ["profile", "email"],
+  })
+);
+
+// callback route for google to redirect to
+router.get(
```

Comment:
Define all these routes in the controllers by making a new file of googleAuth.js, same as we have done for handling url controllers.

Reply:
Okay

Reply:
Changes made as requested